src/job/notify-open-mrs.job.ts

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable import/no-cycle */
+/* eslint-disable no-unused-vars */
+import { BotkitMessage } from 'botkit';
+import * as moment from 'moment';
+
+import logger from '../helpers/Logger';
+import { MergeRequest, IChannelMergeRequests } from '../api/mongo';
+import { service as slack, factory as slackFactory } from '../api/slack';
+/* eslint-enable no-unused-vars */
+
+const { NOTIFY_OPEN_MRS_CRON, NOTIFY_OPEN_MRS_DELAYED_IN_HOURS } = process.env;
+
+const fetchDelayedMRs = (): Promise<IChannelMergeRequests[]> => {
+    const cutDate = moment().subtract(Number(NOTIFY_OPEN_MRS_DELAYED_IN_HOURS), 'hours').toDate();
+
+    return MergeRequest.aggregate()
+        .match({ 'done': false, 'added.at': { '$gte': cutDate } })
+        .group({ '_id': '$slack.channel.id', 'mrs': { '$push': '$$ROOT' } })
+        .exec();
+};
+
+const notifyChannel = (channelMrs: IChannelMergeRequests) => {
+    const message = slackFactory.generateDelayedMergeRequestsMessage(channelMrs);
+
+    // eslint-disable-next-line no-underscore-dangle
+    return slack.mergeDelayed({ 'channel': channelMrs._id } as BotkitMessage, message);
+};
+
+const notifyOpenMrsjob = {
+    'when': NOTIFY_OPEN_MRS_CRON,
+    'function': async function notifyOpenMrs() {
+        logger.info('[notifyOpenMrs] Starting job');
+
+        const openMRs = await fetchDelayedMRs();
+
+        logger.info(`[notifyOpenMrs] Got ${openMRs.length} mrs`);
+
+        if (openMRs.length === 0) {
+            logger.info('[notifyOpenMrs] Stopping job');
+            return this.stop();
```

Comment:
Não pode ser só um return ao invés de this.stop?

Reply:
Esse método do `function` deveria ser mais "independente", não estar acoplado ao cron, poderia ser até uma arrow function.

Reply:
ele é function para manter o contexto, é parte do CronJob. se eu usar uma function eu posso chamar o this.stop(), que faz com que o job não seja mais executado. isso garante que o job não fique rodando infinitamente sem mrs abertos na base. eu dou stop aqui e start quando alguém adiciona um mr na lista

Reply:
Entendi o motivo e gostei.
Mas vc percebe que vc só consegue chamar essa mensagem a partir de um cronjob?
Se quiser colocar essa msg num comando, não funciona.

Reply:
To viajando, não pensei muito sobre, mas podia dar um throw e numa função do job parar o job no catch.

Reply:
concordo, vou ajustar